src/test/java/com/diffmin/AppTest.java

```diff
@@ -194,17 +196,16 @@ void should_apply_mix_operation_patches(TestResources sources) throws Exception
 
     /** Scanner for checking source file of each element. */
     static class ElementSourceFileChecker extends CtScanner {
-        private final Set<String> newRevisionPathStrings;
+        private final List<CtElement> newRevisions;
 
-        ElementSourceFileChecker(Set<String> newRevisionPathStrings) {
-            this.newRevisionPathStrings = newRevisionPathStrings;
+        ElementSourceFileChecker(List<CtElement> newRevisions) {
+            this.newRevisions = newRevisions;
         }
 
         @Override
         public void scan(CtElement element) {
-            if (!skipAssertionCheck(element, newRevisionPathStrings)) {
-                String elementPathString = element.getPath().toString();
-                if (newRevisionPathStrings.contains(elementPathString)
+            if (!skipAssertionCheck(element, newRevisions)) {
+                if (newRevisions.stream().anyMatch(modifiedElement -> modifiedElement == element)
```

Comment:
I was expecting `contains` to have an option to check via `==` but couldn't find it. Had to resort to `anyMatch`.

Reply:
You can use a set collection based around identity.

```java
    static class ElementSourceFileChecker extends CtScanner {
        private final Set<CtElement> newRevisions;

        ElementSourceFileChecker(List<CtElement> newRevisions) {
            this.newRevisions = Collections.newSetFromMap(new IdentityHashMap<>());
            this.newRevisions.addAll(newRevisions);
        }

[...]
```

```suggestion
                if (newRevisions.contains(element)
```